internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -620,6 +608,14 @@ func (r *BMACReconciler) detachedValue(bmh *bmh_v1alpha1.BareMetalHost) (string,
 }
 
 func (r *BMACReconciler) ensureBMHDetached(log logrus.FieldLogger, bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent) reconcileResult {
+	currentValue, haveAnnotation := bmh.ObjectMeta.Annotations[BMH_DETACHED_ANNOTATION]
+
+	// Don't add the annotation if it doesn't exist, but if it does ensure it has the correct value until the user removes it
```

Comment:
Now that assisted won't detach the BMH, users might choose to do it explicitly, unsure we want to replace the value in such case, why not leave it as is?

Reply:
Because otherwise the process for deleting a BMH to remove a node while it is detached will fail for hosts that were previously detached.